libs/environment/src/hooks/use-environment-errors.spec.tsx

```diff
@@ -0,0 +1,199 @@
+jest.mock('../utils/apollo-client');
+
+jest.mock('react-dom', () => ({
+  ...jest.requireActual('react-dom'),
+  createPortal: (node: ReactNode) => node,
+}));
+
+global.fetch = jest.fn();
+
+const MockWrapper = (props: ComponentProps<typeof EnvironmentProvider>) => {
+  return <EnvironmentProvider {...props} />;
+};
+
+const MOCK_HOST = 'https://vega.host/query';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+const mockEnvironmentState = {
+  VEGA_URL: 'https://vega.xyz',
+  VEGA_ENV: Networks.TESTNET,
+  VEGA_CONFIG_URL: 'https://vega.xyz/testnet-config.json',
+  VEGA_NETWORKS: {
+    TESTNET: 'https://testnet.url',
+    STAGNET: 'https://stagnet.url',
+    MAINNET: 'https://mainnet.url',
+  },
+  ETHEREUM_PROVIDER_URL: 'https://ether.provider',
+  ETHERSCAN_URL: 'https://etherscan.url',
+  GIT_BRANCH: 'test',
+  GIT_ORIGIN_URL: 'https://github.com/test/repo',
+  GIT_COMMIT_HASH: 'abcde01234',
+  GITHUB_FEEDBACK_URL: 'https://github.com/test/feedback',
+  setNodeSwitcherOpen: noop,
+  networkError: undefined,
+};
+
+const MOCK_DURATION = 76;
+
+window.performance.getEntriesByName = jest
+  .fn()
+  .mockImplementation((url: string) => [
+    {
+      entryType: 'resource',
+      name: url,
+      startTime: 0,
+      toJSON: () => ({}),
+      duration: MOCK_DURATION,
+    },
+  ]);
+
+function setupFetch(
+  configUrl: string = mockEnvironmentState.VEGA_CONFIG_URL,
+  hosts?: string[]
+) {
+  return (url: RequestInfo) => {
+    if (url === configUrl) {
+      return Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve({ hosts: hosts || [MOCK_HOST] }),
+      } as Response);
+    }
+
+    return Promise.resolve({
+      ok: true,
+    } as Response);
+  };
+}
+
+beforeEach(() => {
+  // @ts-ignore: typescript doesn't recognize the mock implementation
+  global.fetch.mockImplementation(setupFetch());
+
+  window.localStorage.clear();
+
+  // @ts-ignore allow adding a mock return value to mocked module
+  createClient.mockImplementation(() => createMockClient());
+
+  process.env['NX_VEGA_URL'] = mockEnvironmentState.VEGA_URL;
+  process.env['NX_VEGA_ENV'] = mockEnvironmentState.VEGA_ENV;
+  process.env['NX_VEGA_CONFIG_URL'] = mockEnvironmentState.VEGA_CONFIG_URL;
+  process.env['NX_VEGA_NETWORKS'] = JSON.stringify(
+    mockEnvironmentState.VEGA_NETWORKS
+  );
+  process.env['NX_ETHEREUM_PROVIDER_URL'] =
+    mockEnvironmentState.ETHEREUM_PROVIDER_URL;
+  process.env['NX_ETHERSCAN_URL'] = mockEnvironmentState.ETHERSCAN_URL;
+  process.env['NX_GIT_BRANCH'] = mockEnvironmentState.GIT_BRANCH;
+  process.env['NX_GIT_ORIGIN_URL'] = mockEnvironmentState.GIT_ORIGIN_URL;
+  process.env['NX_GIT_COMMIT_HASH'] = mockEnvironmentState.GIT_COMMIT_HASH;
+  process.env['NX_GITHUB_FEEDBACK_URL'] =
+    mockEnvironmentState.GITHUB_FEEDBACK_URL;
+});
+
+describe('throws error', () => {
+  beforeEach(() => {
+    // @ts-ignore: typescript doesn't recognize the mock implementation
```

Comment:
just moving some tests in a separate test suite. we are updating`process.env` in each test case and they are async. when I ran them together with the tests from `use-environment` some of them would fail. this fixes this issue.